packages/excalidraw/components/App.tsx

```diff
@@ -3554,7 +3556,9 @@ class App extends React.Component<AppProps, AppState> {
       }
 
       if (sceneData.elements) {
-        this.scene.replaceAllElements(sceneData.elements);
+        this.scene.replaceAllElements(
+          restoreFractionalIndices(sceneData.elements),
```

Comment:
This is questionable, since our remote elements got already restored indices (as part of restore) & were re-ordered (in reconciliation). Therefore for our multiplayer cases this is unnecessary. This means, that for most of our cases it adds one meaningless loop over all the elements. Other options we have:
* add `isRemoteUpdate` flag to `updateScene` to distinguish between remote and local updates which go through a different process and call `restoreFractionalIndices` only on local udpates
* force the host apps and local `updateScene` cases to always call `restoreElements` (which essentially calls `restoreFractionalIndices`) - this sounds weird, but goes in line with the logic of expecting already reconciled elements in `updateScene`
* leave as it is and potentially refactor away once we will be exchanging just changes between clients (or whatever else we will be doing)

For simplicity I would go with the last option for now, as none of them is ideal.